cmake/platform/platform.cmake.template

```diff
@@ -63,8 +60,12 @@ set(CMAKE_CXX_FLAGS
 
 # STEP 4: Specify that a thread package should be searched in the toolchain
 #         directory. NOTE: when running without threads, remove this line.
-message(STATUS "Requiring thread library")
-FIND_PACKAGE ( Threads REQUIRED )
+#         Here there is a check for the using baremetal scheduler
+if (NOT DEFINED FPRIME_USE_BAREMETAL_SCHEDULER)
```

Comment:
To simplify the toolchain file a bit, can we have the toolchain set `FPRIME_USE_BAREMETAL_SCHEDULE` and then have the OS platform file require Threads for that platform if neccessary?

Reply:
The use of this scheduler is an OS related item, so it belongs in the platform file per our previous discussion on the new organization.

Reply:
Oh I can't read, I thought this was that toolchain template. Don't mind me 😄